File: Python/Implmentation/4-3.py
#check how many ways knights can go
#EX) input: a1 output: 2, since it can only go to c2 or b3

input_data = input()
row = int(input_data[1])
col = int(ord(input_data[0])-int(ord('a'))+1)

dx = [1,-1,2,2,1,-1,-2,-2]
dy = [2,2,1,-1,-2,-2,1,-1]

res = 0
for i in range(8):
    if (8 >= row+dx[i] >= 1 and 8 >= col+dy[i] >= 1):
        res += 1

print(res)

File: C++/Implementation/4-3.cpp
/*
check how many ways knights can go
EX) input: a1 output: 2, since it can only go to c2 or b3
*/

#include <bits/stdc++.h>

using namespace std;

string n;

int dx[] = {1,-1,2,2,1,-1,-2,-2};
int dy[] = {2,2,1,-1,-2,-2,1,-1};

int main(){
    cin >> n;
    int row = n[1] - '0';
    int col = n[0] - 'a' + 1;

    int res = 0;

    for (int i = 0; i < 8 ; i++){
        int nextrow = row+dx[i];
        int nextcol = col+dy[i];
        if (1 <= nextrow && nextrow <= 8 && 1 <= nextcol && nextcol <= 8){
            res++;
        }
    }
    
    cout << res << '\n';
}